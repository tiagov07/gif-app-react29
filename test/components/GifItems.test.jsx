import {GifItem} from  "../../src/components/GifItem"
import { render, screen } from "@testing-library/react"


describe('Test en GifsItem', () => {
    const title = "title"
    const url = "http://localhost/url"
    test('must match with the snapshot', () => {

        const {container} = render( <GifItem title={ title } url={ url }/> )

        expect( container ).toMatchSnapshot()
    })

    test('must show image with the correct url and alt', () => {

        render( <GifItem title={ title } url={ url }/> )

        // screen.debug()

        // expect( screen.getByRole('img').src).toBe( url )
        // expect( screen.getByRole('img').alt).toBe( title )

        const { alt, src } = screen.getByRole('img')
        expect( src ).toBe( url)
        expect( alt ).toBe( alt)
    })

    test('should show the component title', () => {

        render( <GifItem title={ title } url={ url }/> )

        expect( screen.getByText( title )).toBeTruthy()
    })


})