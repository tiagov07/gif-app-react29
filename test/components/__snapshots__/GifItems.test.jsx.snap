// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en GifsItem must match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="http://localhost/url"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
